src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -0,0 +1,271 @@
+        /// <param name="count">The number of unique items that the returned ICollection contains.</param>
+        /// <returns>An instance of an ICollection that can be used for testing.</returns>
+        protected virtual ICollection NonGenericICollectionFactory(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory();
+            AddToCollection(collection, count);
+            return collection;
+        }
+
+        protected virtual bool DuplicateValuesAllowed { get { return true; } }
+        protected virtual bool IsReadOnly { get { return false; } }
+        protected virtual bool NullAllowed { get { return true; } }
+        protected virtual bool ExpectedIsSynchronized { get { return false; } }
+        protected virtual IEnumerable<object> InvalidValues { get { return new object[] { }; } }
+        protected abstract void AddToCollection(ICollection collection, int numberOfItemsToAdd);
+
+        #endregion
+
+        #region IEnumerable Helper Methods
+
+        protected override int WaysToModify { get { return 0; } }
+
+        protected override IEnumerable NonGenericIEnumerableFactory(int count)
+        {
+            return NonGenericICollectionFactory(count);
+        }
+
+        #endregion
+
+        #region Count
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_Count_Validity(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            Assert.Equal(count, collection.Count);
+        }
+
+        #endregion
+
+        #region IsSynchronized
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_IsSynchronized(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            Assert.Equal(ExpectedIsSynchronized, collection.IsSynchronized);
+        }
+
+        #endregion
+
+        #region SyncRoot
+
+        [Theory]
+        [MemberData("ValidCollectionSizes")]
+        public void ICollection_NonGeneric_SyncRoot(int count)
+        {
+            ICollection collection = NonGenericICollectionFactory(count);
+            var root = collection.SyncRoot; // doesnt throw
+        }
+
+        [Fact]
+        public void SyncRootNonNull()
+        {
+            ICollection collection = NonGenericICollectionFactory(16);
+            Assert.NotNull(collection.SyncRoot);
+        }
+
+        [Fact]
+        public void SyncRootConsistent()
+        {
+            ICollection collection = NonGenericICollectionFactory(16);
+            object syncRoot1 = collection.SyncRoot;
+            object syncRoot2 = collection.SyncRoot;
+            Assert.Equal(syncRoot1, syncRoot2);
+        }
+
+        [Fact]
+        public void SyncRootCanBeLocked()
+        {
+            ICollection collection = NonGenericICollectionFactory(16);
+            lock (collection.SyncRoot)
```

Comment:
lock can be used with any object, so this is just testing that SyncRoot is non-null, which we already tested for, right?